metricbeat/module/golang/expvar/expvar.go

```diff
@@ -0,0 +1,127 @@
+package expvar
+
+import (
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/helper"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+)
+
+const (
+	defaultScheme = "http"
+	defaultPath   = "/debug/vars"
+)
+
+var (
+	hostParser = parse.URLHostParserBuilder{
+		DefaultScheme: defaultScheme,
+		DefaultPath:   defaultPath,
+		PathConfigKey: "metrics_path",
+	}.Build()
+)
+
+// init registers the MetricSet with the central registry.
+// The New method will be called after the setup of the module and before starting to fetch data
+func init() {
+	if err := mb.Registry.AddMetricSet("golang", "expvar", New, hostParser); err != nil {
+		panic(err)
+	}
+}
+
+// MetricSet type defines all fields of the MetricSet
+// As a minimum it must inherit the mb.BaseMetricSet fields, but can be extended with
+// additional entries. These variables can be used to persist data or configuration between
+// multiple fetch calls.
+type MetricSet struct {
+	mb.BaseMetricSet
+	http            *helper.HTTP
+	customKeys      []string
+	memstatsEnabled bool
+}
+
+// New create a new instance of the MetricSet
+// Part of new is also setting up the configuration by processing additional
+// configuration entries if needed.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+
+	logp.Warn("EXPERIMENTAL: The golang expvar metricset is experimental")
+
+	config := struct {
+		ExpvarMemstatsEnabled bool `config:"expvar_memstats_enabled"`
+	}{}
+
+	if err := base.Module().UnpackConfig(&config); err != nil {
+		return nil, err
+	}
+
+	return &MetricSet{
+		BaseMetricSet:   base,
+		http:            helper.NewHTTP(base),
+		memstatsEnabled: config.ExpvarMemstatsEnabled,
+	}, nil
+}
+
+// Fetch methods implements the data gathering and data conversion to the right format
+// It returns the event which is then forward to the output. In case of an error, a
+// descriptive error must be returned.
+func (m *MetricSet) Fetch() (common.MapStr, error) {
+
+	json, err := m.http.FetchJSON()
+
+	if err != nil {
+		logp.Err(err.Error())
+	}
+
+	event := common.MapStr{
+		"cmdline": json["cmdline"],
+	}
+
+	if m.memstatsEnabled {
```

Comment:
This should be its own metricset. I even thing that is the one we should start with.